examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,211 @@
+/// A Ratatui example that demonstrates the Elm architecture with a basic list - detail
+/// application.
+///
+/// This example runs with the Ratatui library code in the branch that you are currently
+/// reading. See the [`latest`] branch for the code which works with the most recent Ratatui
+/// release.
+///
+/// [`latest`]: https://github.com/ratatui/ratatui/tree/latest
```

Comment:
It would be good to summarize how this example demonstrates TEA here (also, use //! instead of ///)